staging/cse/windows/configfunc.tests.ps1

```diff
@@ -104,4 +104,73 @@ Describe 'Resize-OSDrive' {
             Assert-MockCalled -CommandName "Set-ExitCode" -Exactly -Times 1 -ParameterFilter { $ExitCode -eq $global:WINDOWS_CSE_ERROR_RESIZE_OS_DRIVE }
         }
     }
+}
+
+Describe 'Config-CredentialProvider' {
+    BeforeEach {
+        $global:KubeDir = "$PSScriptRoot"
+        $CredentialProviderConfPATH=[Io.path]::Combine("$global:KubeDir", "credential-provider-config.yaml")
+    }
+
+    AfterEach {
+        Remove-Item -Path $CredentialProviderConfPATH
+    }
+
+    Context 'CustomCloudContainerRegistryDNSSuffix is empty' {

```

Comment:
Could we store the test data as we did for azurecni https://github.com/Azure/AgentBaker/tree/master/staging/cse/windows/azurecnifunc.tests.suites

Reply:
It is more readable.

Reply:
Sure, wanted to use cni one, but considering the scenario is not not much and the content is little, so I used here-string instead.
Let me update the test code.

Reply:
I just found it hard to handle the kubeDir var in the config yaml file. It's shared both in both test data and in the code.

Reply:
Fixed by passing kubeDir  as a variable and use relative path for azure.json in testdata